modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -4064,13 +4058,16 @@ getFormatParameters <- function(rawOnlyFormat, customFormatSettings, inputFormat
 }
 
 getProtocolEndpointData <- function(protocol) {
+  library(dplyr)
   # Collects all the endpoint values in the protocol object and returns a data frame of them
 
   # First, we create an empty dataframe for the endpoint data associated with the protocol 
   protocolEndpointDataFrame <- data.frame(
     columnName=character(),
     units=character(),
-    dataType=character()
+    dataType=character(),
+    conc = character(),
+    concUnits = character()
```

Comment:
Are we missing Time & Time Units here?

Reply:
I meant to ask - I wasn't sure whether or not to include Time/Time Units initially. Can definitely add them here and the rest of the logic!